doc/config-generic-update-rollback/SONiC_Generic_Config_Update_and_Rollback_Design.md

```diff
@@ -25,51 +25,87 @@
   * [2.1 Target Deployment Use Cases](#21-target-deployment-use-cases)
   * [2.2 Functional Description](#22-functional-description)
     + [2.2.1 Apply-Patch](#221-apply-patch)
-      - [Stage-1 Patch Ordering](#stage-1-patch-ordering)
-      - [Stage-2 Identifying Services to Restart](#stage-2-identifying-services-to-restart)
-      - [Stage-3 Add delay after table updates for listening services to absorb changes](#stage-3-add-delay-after-table-updates-for-listening-services-to-absorb-changes)
-      - [Stage-4 Apply patch, service restarts and delays](#stage-4-apply-patch--service-restarts-and-delays)
-      - [Stage-5 Verify patch update](#stage-5-verify-patch-update)
+      - [Stage-1 JSON Patch Verification](#stage-1-json-patch-verification)
+      - [Stage-2 JSON Patch Ordering](#stage-2-json-patch-ordering)
+      - [Stage-3 Applying list of JsonChanges in order](#stage-3-applying-list-of-jsonchanges-in-order)
+      - [Stage-4 Post-update validation](#stage-4-post-update-validation)
       - [Fail-safe Action](#fail-safe-action)
       - [Logging](#logging)
     + [2.2.2 Checkpoint](#222-checkpoint)
       - [Stage-1 Get current ConfigDB JSON config](#stage-1-get-current-configdb-json-config)
       - [Stage-2 Save JSON config](#stage-2-save-json-config)
-    + [2.2.3 rollback](#223-rollback)
+    + [2.2.3 Rollback](#223-rollback)
       - [Stage-1 Get current ConfigDB JSON config](#stage-1-get-current-configdb-json-config-1)
       - [Stage-2 Get checkpoint JSON config](#stage-2-get-checkpoint-json-config)
       - [Stage-3 Generate the diff as JsonPatch between current config and checkpoint](#stage-3-generate-the-diff-as-jsonpatch-between-current-config-and-checkpoint)
       - [Stage-4 Apply-Patch](#stage-4-apply-patch)
       - [Stage-5 Verify config rollback](#stage-5-verify-config-rollback)
       - [Fail-safe Action](#fail-safe-action-1)
       - [Logging](#logging-1)
+    + [2.2.4 Replace](#224-replace)
+      - [Stage-1 Get current ConfigDB JSON config](#stage-1-get-current-configdb-json-config-2)
```

Comment:
Should we do it at stage 3, after validating the target config?

Reply:
Good idea

Reply:
updated locally

Reply:
updated